R/ggplotMDS.R

```diff
@@ -96,105 +139,209 @@ ggplotMDS <- function(DGEdata,
                 }
             }
         }
-    } else if (is.null(labels)) {
-        addLabels <- FALSE
     }
 
-    assertthat::assert_that(class(DGEdata) %in% c("DGEobj", "DGEList", "matrix"),
-                            msg = "DGEdata must be of class 'DGEList', 'DGEobj', or 'matrix'.")
-    assertthat::assert_that(plotType %in% c("canvasXpress", "ggplot"),
-                            msg = "Plot type must be either canvasXpress or ggplot.")
-
-    if ("DGEobj" %in% class(DGEdata)) {
-        DGEdata <- DGEobj::getItem(DGEdata, "DGEList")
+    if (missing(sizeBy)) {
+        if (!missing(symSize) & any(is.null(symSize), length(symSize) != 1, !is.numeric(symSize))) {
+            warning("symSize must be a singular numeric value. Assigning default value 10.")
+            symSize <- 10
+        }
     }
 
-    assertthat::assert_that(!missing(colorBy),
-                            length(colorBy) == ncol(DGEdata),
-                            msg = "colorBy must be specified and should be the length of the number of columns in DGEdata.")
-    if (!missing(shapeBy)) {
-        assertthat::assert_that(length(shapeBy) == ncol(DGEdata),
-                                msg = "shapeBy should be the length of the number of columns in DGEdata.")
+    if (plotType == "canvasxpress") {
+
+        if (missing(shapeBy)) {
+            if (!missing(symShape) & any(length(symShape) != 1, !.is_valid_symbolShapes_cxplot(symShape))) {
+                warning("symShape must be a singular value of class 'character'. Assigning default value 'circle'.")
+                symShape <- "circle"
+            }
+            shapes <- symShape
+        } else {
+            shapes <- .get_valid_symbolShapes_cxplot()[1:8]
+        }
+
+        intercept_flag <- FALSE
+        if (!missing(hlineIntercept)) {
+            if (!is.null(hlineIntercept)) {
```

Comment:
as there is no else for the second if, code can be reduced by combining both ifs with **&&**